synapse/state.py

```diff
@@ -261,15 +274,30 @@ def compute_event_context(self, event, old_state=None):
                 context.prev_group = entry.prev_group
                 context.delta_ids = dict(entry.delta_ids)
                 context.delta_ids[key] = event.event_id
+
+            context.state_group = yield self.store.store_state_group(
+                event.event_id,
+                event.room_id,
+                prev_group=entry.prev_group,
```

Comment:
why is this `entry.prev_group` rather than `context.prev_group` as calculated above? (any idea when entry.state_group is falsy so we do the prev_group thing?)

Reply:
generally the logic in `compute_event_context` is mysterious  :/

Reply:
> why is this entry.prev_group rather than context.prev_group as calculated above? 

That was a typo tbh.

> (any idea when entry.state_group is falsy so we do the prev_group thing?)

`entry.state_group` is falsey if the resolved state doesn't already have a state group. This would typically happen when the prev events for the event point to two different state groups.

> generally the logic in compute_event_context is mysterious :/

I've tried to add some comments